src/helpers/messages/delete_message.ts

```diff
@@ -1,25 +1,28 @@
 import { botID } from "../../bot.ts";
+import { cacheHandlers } from "../../cache.ts";
 import { RequestManager } from "../../rest/request_manager.ts";
-import { Message } from "../../structures/mod.ts";
+import { Errors } from "../../types/errors.ts";
 import { endpoints } from "../../util/constants.ts";
 import { requireBotChannelPermissions } from "../../util/permissions.ts";
 import { delay } from "../../util/utils.ts";
 
-/** Delete a message */
+/** Delete a message with the channel id and message id only. */
 export async function deleteMessage(
-  message: Message,
+  messageID: string,
   reason?: string,
   delayMilliseconds = 0,
 ) {
+  const message = await cacheHandlers.get("messages", messageID);
+  if (!message) throw new Error(Errors.MESSAGE_NOT_FOUND);
```

Comment:
A vast majority of the time u wont have the message cached. We should not throw this.

```ts
if (message) {
 // checks here
 if (message.author.id !== botID){}
}

```

Reply:
We are also using `message.channelID` property... :thinking: